client/my-sites/checkout/composite-checkout/test/wp-checkout-order-summary.tsx

```diff
@@ -24,17 +45,73 @@ import {
 	getBasicCart,
 	convertProductSlugToResponseProduct,
 } from './util';
-import type { ResponseCart } from '@automattic/shopping-cart';
+import type { ResponseCart, ResponseCartProduct } from '@automattic/shopping-cart';
 import type { FC } from 'react';
 
+type translateType = ReturnType< typeof useTranslate >;
+
+const identity = ( x: string ) => x;
+
 const mockConfig = config as unknown as { isEnabled: jest.Mock };
 jest.mock( '@automattic/calypso-config', () => {
 	const mock = () => '';
 	mock.isEnabled = jest.fn();
 	return mock;
 } );
 
-const identity = ( x: string ) => x;
+const productSlugs = [
+	PRODUCT_JETPACK_ANTI_SPAM_MONTHLY,
+	PRODUCT_JETPACK_ANTI_SPAM,
+	PRODUCT_JETPACK_BACKUP_T1_MONTHLY,
+	PRODUCT_JETPACK_BACKUP_T1_YEARLY,
+	PRODUCT_JETPACK_BOOST_MONTHLY,
+	PRODUCT_JETPACK_BOOST,
+	PLAN_JETPACK_COMPLETE_MONTHLY,
+	PLAN_JETPACK_COMPLETE,
+	PRODUCT_JETPACK_SCAN_MONTHLY,
+	PRODUCT_JETPACK_SCAN,
+	PRODUCT_JETPACK_SEARCH_MONTHLY,
+	PRODUCT_JETPACK_SEARCH,
+	PLAN_JETPACK_SECURITY_T1_MONTHLY,
+	PLAN_JETPACK_SECURITY_T1_YEARLY,
+	PRODUCT_JETPACK_SOCIAL_BASIC_MONTHLY,
+	PRODUCT_JETPACK_SOCIAL_BASIC,
+	PRODUCT_JETPACK_SOCIAL_ADVANCED_MONTHLY,
+	PRODUCT_JETPACK_SOCIAL_ADVANCED,
+	PRODUCT_JETPACK_VIDEOPRESS_MONTHLY,
+	PRODUCT_JETPACK_VIDEOPRESS,
+];
+
+const nonFeatureListProductSlugs = [
+	PRODUCT_JETPACK_BACKUP_T2_MONTHLY,
+	PRODUCT_JETPACK_BACKUP_T2_YEARLY,
+	PLAN_JETPACK_SECURITY_T2_MONTHLY,
+	PLAN_JETPACK_SECURITY_T2_YEARLY,
+];
+
+const allProducts = productSlugs.reduce( ( productList: ResponseCartProduct[], currentProduct ) => {
+	productList.push( convertProductSlugToResponseProduct( currentProduct ) );
+	return productList;
+}, [] );
+
+const allNonFeatureListProducts = nonFeatureListProductSlugs.reduce(
+	( productList: ResponseCartProduct[], currentProduct ) => {
+		productList.push( convertProductSlugToResponseProduct( currentProduct ) );
+		return productList;
+	},
+	[]
+);
```

Comment:
I'd recommend using `.map` instead of `.reduce` to achieve the behavior you're aiming for. `.map` will create a new array with only the values you returned from each iteration, instead of the original values.

```suggestion
const allProducts = productSlugs.map( ( currentProduct ) => {
	return convertProductSlugToResponseProduct( currentProduct );
} );

const allNonFeatureListProducts = nonFeatureListProductSlugs.map( ( currentProduct ) => {
	return convertProductSlugToResponseProduct( currentProduct );
} );
```

Reply:
You're right, that does look cleaner